nixos/modules/programs/gnupg.nix

```diff
@@ -55,6 +79,15 @@ in
   };
 
   config = mkIf cfg.agent.enable {
+    programs.gnupg.agent.pinentryFlavour = mkDefault defaultPinentryFlavour;
+
+    systemd.user.services.gpg-agent = mkIf (cfg.agent.pinentryFlavour != null) {
```

Comment:
this only enables the gpg-agent if `pinentryFlavour` is not null, while we want to consider the enable option for that, and the flavour only for the `--pinentry-program` argument, I guess.

Reply:
Added a comment that this just overrides the `ExecStart` from the upstream units taken from `gnupg` package. It's unfortunately not possible to pass this via an environment variable.